### PR TITLE
adding doc for partition table reading

### PR DESCRIPTION
**Pull request by zhexuany**
In this PR, we add doc for reading from partition table in TiDB.

### CODE REVIEW

File: docs/userguide.md [hunk 1]
```diff
@@ -294,6 +294,9 @@ Currently you could adjust these configs in your spark.conf file.
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | Whether to load statistics info automatically during database mapping. |
 
+## Reading partition table from in TiDB
+Currently, only range partition table is limited supported.
```
Comment (marsishandsome) [inline docs/userguide.md]: which `partition function` do we support?

File: docs/userguide.md [hunk 2]
```diff
@@ -294,6 +294,9 @@ Currently you could adjust these configs in your spark.conf file.
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | Whether to load statistics info automatically during database mapping. |
 
+## Reading partition table from in TiDB
```
Comment (marsishandsome) [inline docs/userguide.md]: delete `in`

File: docs/userguide.md (hunk 1 above)
Comment (marsishandsome) [reply docs/userguide.md]: what if read data from partition table but the function is not supported? throw error?

Comment (zhexuany) [reply docs/userguide.md]: updated. I forget to push the newly commit.

File: docs/userguide.md [hunk 3]
```diff
@@ -294,6 +294,11 @@ Currently you could adjust these configs in your spark.conf file.
 | --------   | -----:   | :----: |
 | spark.tispark.statistics.auto_load | true | Whether to load statistics info automatically during database mapping. |
 
+## Reading partition table from TiDB
+Currently, only range partition table is limited supported. If partition expression having function expression 
+rather than `year` then partition pruning will not be applied. Such scan can be considered full table scan if there is no
+index in the schema.
```
Comment (marsishandsome) [inline docs/userguide.md]: pls delete unnecessary `\n` in one paragraph.

**Pull request merged**